tests/shared/core/test_slot_mappings.py

```diff
@@ -167,3 +167,52 @@ def test_slot_mappings_check_mapping_validity_from_intent():
         mapping=mappings_for_slot[0],
         domain=domain,
     )
+
+
+@pytest.mark.parametrize(
+    "intent, expected",
+    [
+        (["goodbye", "mood_great", "greet"], True),
+        (["a", "b", "c"], False),
+        ([], True),
+        ("", True),
+        ({}, True),
```

Comment:
Based on the previous implementation, the warning is only raised when `mapping.get(INTENT) is not None`. And I retained that.

Do we want to raise the warning when intent is None as well?

Reply:
So `""` and "null" are being interpreted as None.

Is the expected outcome to issue the warning for `""` `[]` and `{}`?

Reply:
You are right, the warning is for a very specific case (when intent key is populated with string or list of string values and the intent(s) are not found in the domain). In this case it should return True for `""` and `[ ]`.
However we should restrict the type of value that can be given to `intent` key via the yaml schema: this should be only string or list of strings or None. In this case yaml validation of the domain file will catch unallowed types like `{ }`. How does that sound?

Reply:
That makes sense. 
But do you think it's within the scope of this ticket?

I'll rather us open another ticket to update the schema validation. What do you think?

Reply:
I'd say it's within scope because we're enabling (fixing) the intent to take on a list of intents and that should be reflected in the schema (which is too permissive I'd say by allowing `Any` data type atm). I'd also double-check if any docs about slot mappings need to be updated.

Reply:
Implementing this is a proving a bit tricky.
From the `pykwalify` docs, all that needs to be done is to change the schema from 
```
intent:
    type: "any"
``` 
to
```
intent:
    type: "seq"
     sequence:
     - type: "str"
``` 

However this restricts intent to always be defined as a list.
So we can't do `intent: deny` anymore, we need to define it like
```
intent:
- deny
```

Or did I miss something? 🤔

Reply:
Oh gotcha, I thought maybe pykwalify has something similar to jsonschema (like "oneOf" pattern, the way implemented for events [here](https://github.com/RasaHQ/rasa/blob/dd139604071715df6d8d04f434a975412f62dfac/rasa/shared/utils/schemas/events.py#L140)). Thanks for trying! We have a ticket in our maintenance backlog to change to using jsonschema instead of yaml schemas in the future 🤞🏻